src/server/modes/opensource/app.ts

```diff
@@ -70,6 +88,76 @@ function initDataLensApp({
     afterAuth.push(xDlContext(), getCtxMiddleware());
 }
 
+function initZitadel({beforeAuth}: {beforeAuth: AppMiddleware[]}) {
+    if (!zitadelUri) {
+        throw new Error('Missing ZITADEL_URI in env');
+    }
+    if (!clientId) {
+        throw new Error('Missing CLIENT_ID in env');
+    }
+    if (!clientSecret) {
+        throw new Error('Missing CLIENT_SECRET in env');
+    }
+    if (!appHostUri) {
+        throw new Error('Missing APP_HOST_URI in env');
+    }
+    if (!cookieSecret) {
+        throw new Error('Missing COOKIE_SECRET in env');
+    }
+
+    passport.use(
+        new OpenIDConnectStrategy(
+            {
+                issuer: zitadelUri,
+                authorizationURL: `${zitadelUri}/oauth/v2/authorize`,
+                tokenURL: `${zitadelUri}/oauth/v2/token`,
+                userInfoURL: `${zitadelUri}/oidc/v1/userinfo`,
+                clientID: clientId,
+                clientSecret: clientSecret,
+                callbackURL: `${appHostUri}/api/auth/callback`,
+                scope: ['openid', 'offline_access', 'urn:zitadel:iam:org:project:id:zitadel:aud'],
+                prompt: '1',
+            },
+            (
+                _issuer: string,
+                _uiProfile: object,
+                _idProfile: object,
+                _context: object,
+                _idToken: string | object,
+                accessToken: string | object,
+                refreshToken: string,
+                _params: unknown,
+                done: VerifyCallback,
+            ) => {
+                if (typeof accessToken !== 'string') {
+                    throw new Error('Incorrect type of accessToken');
+                }
+                return done(null, {accessToken, refreshToken});
+            },
+        ),
+    );
+
+    passport.serializeUser((user: Express.User | null | undefined, done) => {
+        process.nextTick(() => done(null, user));
+    });
+
+    passport.deserializeUser(function (user: Express.User | null | undefined, done): void {
+        process.nextTick(() => done(null, user));
+    });
+
+    beforeAuth.push(
+        cookieSession({
+            secret: cookieSecret,
```

Comment:
Let's add specific `name` for session cookie – `zitadel_session` for example.

Reply:
agree, renamed to zitadelCookieSecret